pgx-pg-sys/src/submodules/panic.rs

```diff
@@ -0,0 +1,424 @@
+/*
+Portions Copyright 2019-2021 ZomboDB, LLC.
+Portions Copyright 2021-2022 Technology Concepts & Design, Inc. <support@tcdi.com>
+
+All rights reserved.
+
+Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+*/
+#![deny(unsafe_op_in_unsafe_fn)]
+#![allow(non_snake_case)]
+
+use std::any::Any;
+use std::cell::Cell;
+use std::fmt::{Display, Formatter};
+use std::hint::unreachable_unchecked;
+use std::panic::{
+    catch_unwind, panic_any, resume_unwind, Location, PanicInfo, RefUnwindSafe, UnwindSafe,
+};
+
+use crate::elog::PgLogLevel;
+use crate::errcodes::PgSqlErrorCode;
+use crate::{pfree, AsPgCStr, MemoryContextSwitchTo};
+
+#[derive(Clone, Debug)]
+pub struct ErrorReportLocation {
+    file: String,
+    line: u32,
+    col: u32,
+}
+
+impl Default for ErrorReportLocation {
+    fn default() -> Self {
+        Self { file: std::string::String::from("<unknown>"), line: 0, col: 0 }
+    }
+}
+
+impl Display for ErrorReportLocation {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}:{}:{}", self.file, self.line, self.col)
+    }
+}
+
+impl From<&Location<'_>> for ErrorReportLocation {
+    fn from(location: &Location<'_>) -> Self {
+        Self { file: location.file().to_string(), line: location.line(), col: location.column() }
+    }
+}
+
+impl From<Option<&Location<'_>>> for ErrorReportLocation {
```

Comment:
I'd probably not impl this and just ask users to `opt.map(ErrorReportLocation::from).unwrap_or_default()` if that's *really* what they want. But I don't feel strongly about it -- it's semantically optional information.